frontend/src/pages/interview/[interviewID].js

```diff
@@ -0,0 +1,135 @@
+/* route to page is /interview/[interviewID] where [interviewID] is a string value of an interview's id */
+import { Component } from 'react'
+import Router from 'next/router'
+import Link from 'next/link'
+import { Container, Button, Row, Col } from 'reactstrap'
+import { getInterviewByID } from '../../utils/api'
+import { ErrorMessage } from '../../components/common'
+import InterviewDetails from '../../components/interview/interviewDetails'
+import VerificationModal from '../../components/verificationModal'
+import Nav from '../../components/nav'
+import Head from '../../components/head'
+import { getKey, deleteInterview } from '../../utils/api'
+
+class InterviewDetailPage extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      errorMessage: '',
+      interview: null,
+      candidateID: '',
+      candidateName: '',
+      verificationModalOpen: false,
+      loading: false
+    }
+  }
+
+  async componentDidMount() {
+    const query = Router.query
+    const { result, success, message } = await getInterviewByID(query.interviewID)
+    if (!success || !result) {
```

Comment:
Personally I would change order of logic here - if we have success and result then set state accordingly, else set the error message. No need for a random return either

```js
if (success && result) {
    this.setState({
       interview: result.interview,
       candidateID: result.candidateID,
       candidateName: result.candidateName
    })
} else {
    this.setState({
        errorMessage: message
    })
}

```

Reply:
I think the random return is nice because it gives the codebase better readability, do you have any other reasons why the return is bad?

Reply:
How does a return give you 'better readability' as opposed to an else condition? The else condition allows the reader to clearly attribute the error message setting when there is some sort of failure with the result.

If the last portion of the code was setting a state for something unrelated to the initial, then the return statement makes sense. But they are part of the same logic flow so the code should be written as such. That makes it much more intuitive for someone to understand.

Reply:
hmmm I guess you have a good point here, thanks for explaining

Reply:
@n3a9 Thanks for mentioning it was a personal preference! You do have a good point in that it is part of the same logic flow.